enclave_manager/avalon_enclave_manager/avalon_enclave_bridge.py

```diff
@@ -178,32 +178,45 @@ def get_enclave_basename():
 
 
 # -----------------------------------------------------------------
-def verify_enclave_info(enclave_info, mr_enclave, originator_public_key_hash):
+def verify_enclave_info(enclave_info, mr_enclave, extended_data):
     """
     Verifies enclave signup info
     - enclave_info is a JSON serialised enclave signup info
       along with IAS attestation report
     - mr_enclave is enclave measurement value
+    - extended_data - MRENCLAVE value of associated WPE
+      in case of KME Enclave and Verification key of associated KME
+      incase of WPE Enclave
     """
-    return enclave.VerifyEnclaveInfo(
-        enclave_info, mr_enclave, originator_public_key_hash)
+    return enclave.VerifyEnclaveInfo(enclave_info, mr_enclave, extended_data)
 
 
 # -----------------------------------------------------------------
-def create_signup_info(originator_public_key_hash, nonce):
+def create_signup_info(nonce, worker_mode=1, extended_data="",
+                       extended_data_signature="", kme_attesation=""):
```

Comment:
This function is generalized too much. kme_attestation has to be passed based on the worker_mode right? Better to split this function to handle different modes, or create interface for this

Reply:
This function doesn't have much changes except calling function with different functions. To avoid lot of duplicate code, i have added all the parameters to cater multiple worker types.